### PR TITLE
Bugfix: Resource paths mapped to `main` site instead of `work` site.

### PR DESCRIPTION
**Pull request by Sharkitty**
## Changelog Description
Fix resource files being mapped to main site instead of work site.

## Testing notes
- Open any workfile with resource files.
- Publish it.
- Use the fab_old technique (or force download last workfile if available).
- Check resource paths, they should be mapped to the `resources` folder in the local workfile folder.
- Check all link paths, they should be unaffected by these changes (pointing to `main` site path).

### CODE REVIEW

File: openpype/hosts/blender/api/utils.py [hunk 1]
```diff
@@ -433,17 +437,52 @@ def make_paths_absolute(source_filepath: Path = None):
...
+                        context_filters={
+                            "asset": get_current_asset_name(),
+                            "family": "workfile",
+                            "task": {"name": get_current_task_name()},
+                        },
+                    )
+                )
+            ),
+            key=lambda r: r["context"]["version"],
+        )
```
Comment (Tilix4) [inline openpype/hosts/blender/api/utils.py]: Isn't it some logic shared with the download last workfile? Can't we make it a common function?

File: openpype/hosts/blender/utility_scripts/make_paths_relative.py [hunk 2]
```diff
@@ -15,19 +22,59 @@
...
+                    context_filters={
+                        "asset": get_current_asset_name(),
+                        "family": "workfile",
+                        "task": {"name": get_current_task_name()},
+                    },
+                )
+            ),
+        ),
+        key=lambda r: r["context"]["version"],
+    )
```
Comment (Tilix4) [inline openpype/hosts/blender/utility_scripts/make_paths_relative.py]: Definitely requires a function...

File: openpype/hosts/blender/utility_scripts/make_paths_relative.py [hunk 3]
```diff
@@ -15,19 +22,59 @@
...
+                        str(
+                            Path(
+                                workdir,
+                                "resources",
+                                Path(datablock.filepath).name,
+                            )
+                        ),
+                        start=workdir,
+                    )
+                    break
```
Comment (Tilix4) [inline openpype/hosts/blender/utility_scripts/make_paths_relative.py:61]: why do you break? What if there are several?

**Review by Tilix4 — CHANGES_REQUESTED** [review]

File: openpype/hosts/blender/utility_scripts/make_paths_relative.py (hunk 3 above)
Comment (Sharkitty) [reply openpype/hosts/blender/utility_scripts/make_paths_relative.py:61]: Several filepaths for the same datablock? O.o

File: openpype/hosts/blender/api/utils.py [hunk 4]
```diff
@@ -433,17 +438,41 @@ def make_paths_absolute(source_filepath: Path = None):
     relative_datablocks = get_datablocks_with_filepath(absolute=False)
     remapped_datablocks = set()
     if source_filepath:
+        # Get last workfile representation
```
Comment (Tilix4) [inline openpype/hosts/blender/api/utils.py]: now the comment becomes useless

File: openpype/pipeline/workfile/path_resolving.py [hunk 5]
```diff
@@ -528,3 +532,21 @@ def create_workdir_extra_folders(
         fullpath = os.path.join(workdir, subfolder)
         if not os.path.exists(fullpath):
             os.makedirs(fullpath)
+
+def get_last_workfile_representation(project_name, asset_name, task_name):
```
Comment (Tilix4) [inline openpype/pipeline/workfile/path_resolving.py:536]: that's pretty cool, can you propose that to OP?

**Review by Tilix4 — CHANGES_REQUESTED** [review]

File: openpype/pipeline/workfile/path_resolving.py (hunk 5 above)
Comment (Sharkitty) [reply openpype/pipeline/workfile/path_resolving.py:536]: Sure thing

**Review by Tilix4 — APPROVED** [review]

**Comment by Sharkitty** [comment]
[PR to the get last workfile representation](https://github.com/ynput/OpenPype/pull/5742)

**Pull request merged**